include/tiramisu/core.h

```diff
@@ -740,6 +740,36 @@ class function
      */
     function(std::string name);
 
+     /**
+     * \brief Generates the automatic communication CPU/GPU.
+     * 
+     */
+    const int  &Automatic_communication(tiramisu::computation* c1,tiramisu::computation* c2) const;
+    /**
+     * \brief Returns a ptr to the first computation in the sched_graph.
+     * \details The computation that have no predecessor
+     */
+    computation * get_first_cpt();
+     /**
+     * \brief Returns a ptr to the last computation in the sched_graph.
+     * \details The computation that have no succesor
+     */
+    computation * get_last_cpt();
+     /**
+     * \brief This map contains the names of the cpu buffers and the pointers of the corresponding gpu buffers. 
+     * \details it is modified when creating a gpu buffer, please have a look at the buffer constructor.
+     */
+    std::map<std::string, tiramisu::buffer *> mapping ;
```

Comment:
Is this a public member? If so, I think it should not be a public member. You should use get/set functions or a higher level function to access it. We want to be able to change the way it is implemented without changing the code in the future.